### PR TITLE
Add array rotation in C

### PR DESCRIPTION
**Pull request by sandhyabhan**
<Issue number #1766 >
Fixes #

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added code for array rotation with test inputs.

- Spaces added between operators and after comma's.

#### Languages Used:
- C
- 

#### Files Added:
- array.c
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Array_Rotation/arrayrotation.c [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        a[i] = a[i+1];
+        a[n] = temp;
+    }
+    }
+
+    for(i=0 ; i<n ; i++)
+    {
+    printf("%d " , a[i]);
+    }
+
```
Comment (raghav-dalmia) [inline Array_Rotation/arrayrotation.c]: Remove extra lines

File: Array_Rotation/arrayrotation.c [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+
+
+
+
+return 0 ;
+}
+
+//Test input
```
Comment (raghav-dalmia) [inline Array_Rotation/arrayrotation.c]: properly comments these lines

File: Array_Rotation/arrayrotation.c [hunk 3]
```diff
@@ -0,0 +1,50 @@
+#include<stdio.h>
+#include<stdlib.h>
+
+int main()
+{
+    int a[1000000] = {0};
+    int i=0 ;
```
Comment (raghav-dalmia) [inline Array_Rotation/arrayrotation.c]: club them in one line

File: Array_Rotation/arrayrotation.c [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    for(i=0 ; i<n ; i++)
+    {
+        scanf("%d" , &a[i]);	// scanning elements
+    }
+
+    for(j=0 ; j<d ; j++)
+    {
+        temp = a[0];
+
+    for(i=0 ; i<n ; i++)
```
Comment (raghav-dalmia) [inline Array_Rotation/arrayrotation.c]: indentation issue

File: Array_Rotation/arrayrotation.c [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    {
+        a[i] = a[i+1];
+        a[n] = temp;
+    }
+    }
+
+    for(i=0 ; i<n ; i++)
+    {
+    printf("%d " , a[i]);
```
Comment (raghav-dalmia) [inline Array_Rotation/arrayrotation.c]: here also

**Comment by somya-kapoor** [comment]
Please update the PR description, use [x] not [x ].

File: Array_Rotation/arrayrotation.c [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        temp = a[0];
+
+        for(i=0 ; i<n ; i++)
+        {
+            a[i] = a[i+1];
+            a[n] = temp;
+        }
+    }
+    for(i=0 ; i<n ; i++)
```
Comment (somya-kapoor) [inline Array_Rotation/arrayrotation.c]: Add space around the operator.

File: Array_Rotation/arrayrotation.c [hunk 7]
```diff
@@ -0,0 +1,35 @@
+#include<stdio.h>
+#include<stdlib.h>
+
+int main()
+{
+    int a[1000000] , int i , int j, int n , int d , int temp ;
+    scanf("%d %d" , &n , &d);                   // n is the number of elements and d is the number of times rotation.
+    for(i=0 ; i<n ; i++)
```
Comment (somya-kapoor) [inline Array_Rotation/arrayrotation.c]: Add space around the operator.

File: Array_Rotation/arrayrotation.c [hunk 8]
```diff
@@ -0,0 +1,37 @@
+#include<stdio.h>
```
Comment (somya-kapoor) [inline Array_Rotation/arrayrotation.c]: Please change the file name to  Array_Rotation/Array_Rotation.c

File: Array_Rotation/arrayrotation.c [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+            a[i] = a[i+1];
+            a[n] = temp;
+        }
+    }
+     //printing the elements of array
+    for(i = 0; i < n; i++)
+    {
+        printf("%d", a[i]);      
+    }
+return 0 ;
```
Comment (somya-kapoor) [inline Array_Rotation/arrayrotation.c]: Please correct the indentation of return

File: Array_Rotation/ArrayRotation.c [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+            a[i] = a[i+1];
+            a[n] = temp;
+        }
+    }
+     //printing the elements of array
+    for(i = 0; i < n; i++)
+    {
+        printf("%d", a[i]);      
+    }
+  return 0 ;
```
Comment (somya-kapoor) [inline Array_Rotation/ArrayRotation.c]: Indentation of return is not correct.

File: Array_Rotation/arrayrotation.c (hunk 8 above)
Comment (somya-kapoor) [reply Array_Rotation/arrayrotation.c]: @sandhyabhan Please change the file name to Array_Rotation.c not ArrayRotation.c

File: Array_Rotation/ArrayRotation.c [hunk 11]
```diff
@@ -0,0 +1,37 @@
+#include<stdio.h>
+#include<stdlib.h>
+int main(void)
```
Comment (somya-kapoor) [inline Array_Rotation/ArrayRotation.c]: Leave a line before int main

File: Array_Rotation/ArrayRotation.c [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+{
+    int a[1000000], int i, int j, int n, int d, int temp ;
+     // n is the number of elements and d is the number of times rotation.
```
Comment (somya-kapoor) [inline Array_Rotation/ArrayRotation.c]: Indentation of comments is not correct.

File: Array_Rotation/ArrayRotation.c [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+    }
+  return 0 ;
+}
+
+//Test input
+INPUT: 5 4 (number of elements and no. of times rotation)
+       1 2 3 4 5 (the input array)
+
+Output: 5 1 2 3 4 
+/
```
Comment (somya-kapoor) [inline Array_Rotation/ArrayRotation.c]: double /

File: Array_Rotation/ArrayRotation.c [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+    }
+    //printing the elements of array
+    for(i = 0; i < n; i++)
+    {
+        printf("%d", a[i]);      
+    }
+   return 0 ;
+}
+
+//Test input
```
Comment (somya-kapoor) [inline Array_Rotation/ArrayRotation.c]: // is used for single-line comments
/*comments*/ iss used for multi line comments.
But you are using // for multi-line. Please change it.

File: Array_Rotation/arrayrotation.c (hunk 8 above)
Comment (sandhyabhan) [reply Array_Rotation/arrayrotation.c]: I tried doing so...but it is showing error that file name already existed.
what should i do?

File: Array_Rotation/ArrayRotation.c [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+    }
+    //printing the elements of array
+    for(i = 0; i < n; i++)
+    {
+        printf("%d", a[i]);      
+    }
+   return 0 ;
+}
+
+/Test input
```
Comment (somya-kapoor) [inline Array_Rotation/ArrayRotation.c]: instead of / use /*

File: Array_Rotation/ArrayRotation.c [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+INPUT: 5 4 (number of elements and no. of times rotation)
+       1 2 3 4 5 (the input array)
+
+Output: 5 1 2 3 4 
+/
```
Comment (somya-kapoor) [inline Array_Rotation/ArrayRotation.c]: use */

File: Array_Rotation/ArrayRotation.c [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        scanf("%d", &a[i]);	
+    }
+    for(j = 0; j < d; j++)
+    {
+        temp = a[0];
+
+        for(i = 0; i < n; i++)
+        {
+            a[i] = a[i+1];
```
Comment (somya-kapoor) [inline Array_Rotation/ArrayRotation.c]: Add space around +

**Comment by somya-kapoor** [comment]
@sandhyabhan Please squash the commits

**Comment by asha15** [comment]
could you change the file name as Array_Rotation.c ? then we are good to merge

**Comment by sandhyabhan** [comment]
I tried doing it earlier too...its showing same file name exists..what can i do?

**Comment by asha15** [comment]
yes. actually it's already implemented 

@jainaman224 what we do now ?

**Comment by sandhyabhan** [comment]
How come? I was alloted this issue by @jainaman224 and have been working on it since then.

**Pull request closed without merging**